### ISSUE TITLE
Materials Duplicating When Using "Edit As Maya Data"

### ISSUE THREAD

**Issue by jimwatson-ta** [Autodesk/maya-usd#3547]
**Describe the bug**
After clicking "Edit As Maya Data" on a UsdMesh, it duplicates the materials in the scene if a material of the same name exists. 

**Steps to reproduce**
1. Create a cube in Maya
2. Assign one lambert material to the top face and one material to the other faces of the cube.
3. Create a new USD layer. 
4. Duplicate the Cube as USD data into the stage. 
5. Edit the USD Cube as Maya Data. 
6. See duplicate materials assigned on the mesh and new materials in the Hypershade.

**Expected behavior**
When hitting the button to "Edit As Maya Data," if the linked materials already exist in the scene, assign that material to the mesh. 

**Specs (if applicable):**
 - Windows 10
 - Codeless Schemas
 - Maya 2024.2
 - MayaUSD 0.26.0

**Comment by wallworm** [comment]
Thank you for the input. What about the case where the material in the USD is actually more updated than the scene material? Do you have a proposal on handling which materials (Keep Scene Materials, Update Scene Materials or Make Duplicate)?

**Comment by jimwatson-ta** [comment]
Good call. I would prefer if the scene materials would be overridden, but I can also see that as a useful option on the Edit As Maya Data Options menu.
